### DISCUSSION TITLE
v0.12.8

### DISCUSSION THREAD

**Discussion by AsamK** [AsamK/signal-cli#1441] [Releases]
**Full Changelog**: https://github.com/AsamK/signal-cli/compare/v0.12.7...v0.12.8

<hr /><em>This discussion was created from the release <a href='https://github.com/AsamK/signal-cli/releases/tag/v0.12.8'>v0.12.8</a>.</em>

**Comment by nkuttler** [comment]
File signal-cli-0.12.7-Linux-native.tar.gz  contains version 0.12.8, not matching the archive filename.

**Comment by orazioedoardo** [comment]
Thanks, this fixed Signal server pulling the plug at ~20:40 UTC yesterday, feb 5.

**Reply by Techie007L** [reply]
Yes, I had just updated a few days ago to signal-cli-0.12.7 due to them pulling the plug on signal-cli-0-11.4 on 1/31/2024.  Now on 2/5/2024 (not even a week later) they've pulled the plug on the latest signal-cli-0.12.7 too!  Hopefully this is not a sign of things to come.  I wonder if they're displeased with things like not receiving messages (mine is only used to send notifications)—currently looking into switching it to daemon mode.  If this is the case, then perhaps @AsamK should look into disabling the one-off message sending mode in a future release to prevent such "abuse" (also not a bad idea considering that you can't use both modes at the same time; the "one-off" mode can't send a message and keeps running in the background until the daemon process is stopped).  Switching to daemon mode should result in faster notifications too, since starting the Java virtual machine, initializing Signal, and connecting to Signal servers takes 5-10 seconds each time (and maybe they don't like that either, the regular connecting+disconnecting, instead of just staying connected and sending data from time to time).  Sending and receiving a message from the daemon happens <2 seconds.

**Reply by LFrank2021** [reply]
I was wondering the exact same thing.
If this will become the new norm signal is no longer usable for me.
Is there any information why, when signal is pulling an API-version?

**Reply by orazioedoardo** [reply]
> If this will become the new norm signal is no longer usable for me.

It's already the norm that Signal knocks out old versions but usually signal-cli updates the user agent often enough that there are no interruptions.

**Reply by LFrank2021** [reply]
I wonder why the 0.10.x and 0.11.x were working for several years.
I just upgraded to 0.12.7 a few days back and already it got pulled again.

**Comment by aggr0lite** [comment]
How does a complete upgrade procedure usually look like? After it stopped working yesterday, I today just installed signal-cli again as if it was a fresh install, adjusted the libsignal_jni.so and started the daemon again. 
But I constantly get this error now when receiving messages:

> Envelope from: “xxx” xxx (device: 1) to xxx
> Timestamp: 1707209572616 (2024-02-06T08:52:52.616Z)
> Server timestamps: received: 1707211451456 (2024-02-06T09:24:11.456Z) delivered: 1707211451458 (2024-02-06T09:24:11.458Z)
> Sent by unidentified/sealed sender
> Exception: org.signal.libsignal.protocol.InvalidMessageException: invalid PreKey message: decryption failed (ProtocolInvalidMessageException)

Does an upgrade require a new account registration? Sending messages works as expected this way:

signal-cli -a xxxx send -m "test message" xxxx
